src/App.cpp

```diff
@@ -228,7 +228,7 @@ int MyApp::OnExit()
         m_luaLog->Close();
         // Check to see if we have lua messages
         if (m_luaMessages) {
-            XWordErrorMessage(NULL, _T("Errors occurred.  See log file: %s"), (const wxChar *)GetLuaLogFilename().c_str());
+            XWordErrorMessage(NULL, _T("Errors occurred.  See log file: %s"), GetLuaLogFilename().c_str().AsChar());

```

Comment:
Using mb_str() with a cast definitely doesn't work for me -- at least on windows, wxString is using wstring internally, so when XWordErrorMessage calls wxString::FormatV, we eventually end up at wxVsnprintf, which uses the wide version

The upshot is that for me, "%s" returns gibberish and "%hs" returns the actual message.

From what I can tell, wx_str() fixes this problem so that if , I'm calling wsprintf with a wchar\* instead of a char\* . . . if we were using sprintf, the mb_str and char\* cast would be necessary, but I think since we're calling a wx function, wx_str is the way to go.

If wx_str() doesn't work on your end, we probably have some thinking to do.